lib/fact_check_email_handler.rb

```diff
@@ -39,13 +39,16 @@ def process_message(message)
 
   # takes an optional block to call after processing each message
   def process
+    email_count = 0
     Mail.all(read_only: false, delete_after_find: true) do |message|
-      message.skip_deletion unless process_message(message)
+      message.mark_for_delete = process_message(message)
+      email_count += 1 unless message.is_marked_for_delete?
       begin
         yield(message) if block_given?
       rescue StandardError => e
         GovukError.notify(e)
       end
     end
+    GovukStatsd.gauge("email.count", email_count)
```

Comment:
I think it might be better if this was named something like `unprocessed_email_count` to make it clear that it's the number of emails that weren't able to be processed, i.e. deleted.

Reply:
Technically those emails were processed but not deleted - how about this?
`GovukStatsd.gauge("undeleted_emails.count", undeleted_emails_count)`

Reply:
Yeah, that's good too.

Reply:
I changed the code to follow your naming suggestion. Even though we technically process the emails and "undeleted" would be more accurate, the language around this is all about "processing" the messages (method names, etc), so I think it’s better to remain consistent.